cron_sentry/runner.py

```diff
@@ -152,22 +164,32 @@ def run(self):
             with TemporaryFile() as stderr:
                 try:
                     exit_status = call(self.command, stdout=stdout, stderr=stderr)
+                    last_lines_stdout = self._get_last_lines(stdout)
+                    last_lines_stderr = self._get_last_lines(stderr)
                 except CommandNotFoundError as exc:
                     last_lines_stdout = ''
                     last_lines_stderr = str(exc)
                     exit_status = 127  # http://www.tldp.org/LDP/abs/html/exitcodes.html
-                else:
-                    last_lines_stdout = self._get_last_lines(stdout)
-                    last_lines_stderr = self._get_last_lines(stderr)
 
-                if self.report_all or exit_status != 0:
-                    elapsed = int((time() - start) * 1000)
-                    self.report(exit_status, last_lines_stdout, last_lines_stderr, elapsed)
+                elapsed = int((time() - start) * 1000)
 
                 if not self.quiet:
                     sys.stdout.write(last_lines_stdout)
                     sys.stderr.write(last_lines_stderr)
 
+                if self.report_all:
+                    self.report(exit_status, last_lines_stdout ,last_lines_stderr,elapsed)
+                    return exit_status
+
+                if exit_status != 0:
+                    self.level = logging.ERROR
+                    self.report(exit_status, last_lines_stdout, last_lines_stderr, elapsed)
```

Comment:
won't this report twice if it exits with anything but 0 and has stderr because there's no `return exit_status` in this `if` block?